pkg/apis/pipeline/v1alpha1/build_gcs_resource_test.go

```diff
@@ -0,0 +1,300 @@
+*/
+
+package v1alpha1
+
+import (
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func Test_Invalid_BuildGCSResource(t *testing.T) {
+	testcases := []struct {
+		name             string
+		pipelineResource *PipelineResource
+	}{{
+		name: "no location params",
+		pipelineResource: &PipelineResource{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "buildgcs-resource-with-no-location-param",
+			},
+			Spec: PipelineResourceSpec{
+				Type: PipelineResourceTypeStorage,
+				Params: []Param{{
+					Name:  "NotLocation",
+					Value: "doesntmatter",
+				}, {
+					Name:  "type",
+					Value: "build-gcs",
+				}},
+			},
+		},
+	}, {
+		name: "location param with empty value",
+		pipelineResource: &PipelineResource{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "gcs-resource-with-empty-location-param",
+			},
+			Spec: PipelineResourceSpec{
+				Type: PipelineResourceTypeStorage,
+				Params: []Param{{
+					Name:  "Location",
+					Value: "",
+				}, {
+					Name:  "type",
+					Value: "build-gcs",
+				}},
+			},
+		},
+	}, {
+		name: "no artifactType params",
+		pipelineResource: &PipelineResource{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "buildgcs-resource-with-no-artifactType-param",
+			},
+			Spec: PipelineResourceSpec{
+				Type: PipelineResourceTypeStorage,
+				Params: []Param{{
+					Name:  "Location",
+					Value: "gs://test",
+				}, {
+					Name:  "type",
+					Value: "build-gcs",
+				}},
+			},
+		},
+	}, {
+		name: "artifactType param with empty value",
+		pipelineResource: &PipelineResource{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "gcs-resource-with-empty-location-param",
+			},
+			Spec: PipelineResourceSpec{
+				Type: PipelineResourceTypeStorage,
+				Params: []Param{{
+					Name:  "Location",
+					Value: "gs://test",
+				}, {
+					Name:  "type",
+					Value: "build-gcs",
+				}, {
+					Name:  "ArtifactType",
+					Value: "",
+				}},
+			},
+		},
+	}, {
+		name: "artifactType param with invalid value",
+		pipelineResource: &PipelineResource{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "gcs-resource-with-empty-location-param",
+			},
+			Spec: PipelineResourceSpec{
+				Type: PipelineResourceTypeStorage,
+				Params: []Param{{
+					Name:  "Location",
+					Value: "gs://test",
+				}, {
+					Name:  "type",
+					Value: "build-gcs",
+				}, {
+					Name:  "ArtifactType",
+					Value: "invalid-type",
+				}},
+			},
+		},
+	}, {
+		name: "artifactType param with secrets value",
+		pipelineResource: &PipelineResource{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "gcs-resource-with-secrets",
+			},
+			Spec: PipelineResourceSpec{
+				Type: PipelineResourceTypeStorage,
+				Params: []Param{{
+					Name:  "Location",
+					Value: "gs://test",
+				}, {
+					Name:  "type",
+					Value: "build-gcs",
+				}, {
+					Name:  "ArtifactType",
+					Value: "invalid-type",
+				}},
+				SecretParams: []SecretParam{{
```

Comment:
@tejal29 :  I think I might be missing something
The difference between `artifactType param with invalid value` and `artifactType param with secrets value"` test is secretParams.  how does that make diff ?